testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/MultiFactorAuthenticationTest.java

```diff
@@ -147,6 +147,12 @@ public void testAlternativeCredentials() {
             // like user with user-with-one-configured-otp. However OTP is preferred credential for him, so OTP mechanism will take preference
```

Comment:
Acutally, I just saw this comment.

It seems that the order of credentials did indeed matter for determining the order of authentication options. 
However, AFAIK the user cannot control the order of credentials - only administrators can change the credential order.

I think relying on the order of stored credentials is fragile and very confusing. It would be better to rely on a deterministic order governed by the authentication flow.